File: generator/client/golang/templates/queryx/select_test.go
package queryx

import (
	"testing"

	"github.com/stretchr/testify/require"
)

func TestSelect(t *testing.T) {
	s := NewSelect().Select("users.*").From("users")
	sql, args := s.ToSQL()
	require.Equal(t, `SELECT users.* FROM users`, sql)
	require.Equal(t, []interface{}{}, args)

	sql, args = s.Update().Columns("name", "email").Values("test", "test@example.com").ToSQL()
	require.Equal(t, `UPDATE users SET name = ?, email = ?`, sql)
	require.Equal(t, []interface{}{"test", "test@example.com"}, args)
}

File: generator/client/typescript/templates/queryx/select.test.ts
import { test, expect } from "vitest";
import { newSelect } from "./select";
import { Clause } from "./clause";

test("select", () => {
  let s = newSelect().select("users.*").from("users");
  expect(s.toSQL()).toEqual(["SELECT users.* FROM users", []]);
});

test("select where", () => {
  let s1 = newSelect()
    .select("users.*")
    .from("users")
    .where(new Clause("id = ?", [1]));
  expect(s1.toSQL()).toEqual(["SELECT users.* FROM users WHERE id = ?", [1]]);

  s1.where(new Clause("name = ?", ["test"]));
  expect(s1.toSQL()).toEqual([
    "SELECT users.* FROM users WHERE (id = ?) AND (name = ?)",
    [1, "test"],
  ]);
});
